src/ir/possible-contents.cpp

```diff
@@ -1016,6 +1016,23 @@ struct InfoCollector
     addChildParentLink(curr->ref, curr);
     addChildParentLink(curr->value, curr);
   }
+  void visitStructRMW(StructRMW* curr) {
+    if (curr->ref->type == Type::unreachable) {
+      return;
+    }
+    // Similar to visitStructSet.
+    addChildParentLink(curr->ref, curr);
+    addChildParentLink(curr->value, curr);
```

Comment:
Adding the links is necessary, but custom code must then handle it. See line 2502 for StructGet and Set. I imagine these just need to do both a readFromData and writeToData?

Reply:
My guess is that `writeToData` would do the wrong thing because RMW operations need to combine the value expression with the read data, rather than just take the value and store it somewhere.

I guess I should probably just use `addRoot` to avoid getting in over my head here?

Reply:
Yeah, addRoot is fine - it does seem hard to figure out the value here in general.